stardist_napari/_dock_widget.py

```diff
@@ -202,21 +203,40 @@ def plugin (
         x = get_data(image)
         axes = axes_check_and_normalize(axes, length=x.ndim)
 
-        if not axes.startswith(model._axes_out.replace('C','')):
+        if output_type in (Output.Polys.value,Output.Both.value) \
+           and 'T' in axes and isinstance(model, StarDist3D):
+            raise NotImplementedError('Polyhedra output currently not supported for 3D timelapse data')
+
+        if not axes.replace('T','').startswith(model._axes_out.replace('C','')):
             warn(f"output images have different axes ({model._axes_out.replace('C','')}) than input image ({axes})")
             # TODO: adjust image.scale according to shuffled axes
 
         if norm_image:
             # TODO: address joint vs. channel-separate normalization properly (let user choose)
-            if 'C' not in axes or image.rgb == True:
+            if 'C' not in axes or image.rgb == True or 'T' in axes:
                  # normalize channels jointly
```

Comment:
Normalizing all frames of a timelapse jointly is one option, but I'd say not clearly always what you want to do (e.g. sample bleach). I would rather normalize each frame independently as the default.

Reply:
Sure, we could do that

Reply:
Downside could be that segmented objects are less stable across time.

Reply:
But there's always the option to normalize the image manually before running the plugin.

Reply:
Ideally, one could give the axes to be normalized over (defaulting eg to "XY")

Reply:
> Ideally, one could give the axes to be normalized over (defaulting eg to "XY")

This is a proper solution that needs more work but also adds complexity.

Reply:
I just changed it: https://github.com/stardist/stardist-napari/pull/4/commits/539d4338f0e10c8deb025ccfdace76db1daebb9a

Reply:
> Ideally, one could give the axes to be normalized over (defaulting eg to "XY")

Do we want to do that now or later?

Reply:
How hard would it be?

Reply:
Depends on how we implement it. We could
1. (as you suggested) let the user provide the axes to be normalized over (e.g. "XY"), or we can
2. have check/radio-boxes to choose joint/separate normalization per channel/frame

Option 1 is easier to implement and more generic, but likely confusing for users and more difficult to persist / adapt to the axes of the currently selected image.
Option 2 is more work to implement, but likely easier to understand for users and easy to persist.

Reply:
I slightly tend towards Option 1 as this anyhow will be an expert option...

Reply:
The default normalization axes could be "ZYX", meaning that all spatial dimensions will be jointly normalized, but not extra time or channel dimensions. But what do we do for RGB data? Also, is it confusing that "ZYX" also applies for 2D images (i.e. "Z" is not relevant for a 2D image)?

Reply:
Well, the default axes should be the image axes, as already automatically inferred during initialization, or?

Reply:
No, it's not that simple. It's a preference that can be persisted, or we change it a default every time the image changes. That could be annoying to people who have to change it constantly.

Reply:
Yes, I see your point. Wouldn't Option 2 have the same issue?

Reply:
I've implemented option 1: https://github.com/stardist/stardist-napari/pull/4/commits/7fa9970712ecb385bae5e8778e732d1729901022

Reply:
Nice! Thx!

Reply:
Feature done?